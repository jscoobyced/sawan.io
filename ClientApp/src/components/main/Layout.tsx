import React from 'react';
import { MainContent } from '../../services/Models';
import { Footer } from './Footer';
import { Header } from './Header';
import { SideBar } from './SideBar';

export interface LayoutProps {
    children: React.ReactNode;
    allContent?: MainContent;
    isSignedIn: boolean;
    signIn: (isSignedIn: boolean) => void;
}

export class Layout extends React.Component<LayoutProps> {

    public constructor(props: LayoutProps) {
        super(props);
    }

    public render() {
        if (this.props.allContent == null) {
            return <div />;
        }

        return (
            <main>
                <section>
                    <Header
                        signIn={this.props.signIn}
                        isSignedIn={this.props.isSignedIn}
                        navigationMenuContent={this.props.allContent.navigationMenuContent}
                    />
                    {this.props.children}
                </section>
                <SideBar allContent={this.props.allContent} />
                <Footer footerContent={this.props.allContent.footerContent} />
            </main>);
    }
}
