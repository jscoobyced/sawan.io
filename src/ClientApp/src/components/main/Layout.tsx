import React from 'react';
import { AllContent } from '../../services/Models';
import { Footer } from './Footer';
import { Header } from './Header';
import { SideBar } from './SideBar';

export interface LayoutProps {
    children?: React.ReactNode;
    allContent?: AllContent;
}

export class Layout extends React.Component<LayoutProps> {

    public constructor(props: LayoutProps) {
        super(props);
    }

    public render() {
        if (this.props.allContent == null) {
            return <div></div>;
        }

        return <main>
            <Header navigationMenuContent={this.props.allContent.navigationMenuContent} />
            <section>
                {this.props.children}
            </section>
            <SideBar menuContent={this.props.allContent.menuContent} />
            <Footer />
        </main>;
    }
}
