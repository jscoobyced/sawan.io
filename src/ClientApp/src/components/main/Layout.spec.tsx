import { shallow } from 'enzyme';
import React from 'react';
import { Layout } from './Layout';

test('Layout component is unchanged.', () => {
    const layout = shallow(<Layout />);
    expect(layout).toMatchSnapshot();
});