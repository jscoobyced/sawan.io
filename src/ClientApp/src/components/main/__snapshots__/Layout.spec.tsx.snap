// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout base component is unchanged. 1`] = `<div />`;

exports[`Layout component with content is unchanged. 1`] = `
<main>
  <Header
    navigationMenuContent={
      Object {
        "about": "",
        "applications": "",
        "cryptoCurrency": "",
        "healthMonitor": "",
        "home": "",
        "websiteName": "",
      }
    }
  />
  <section />
  <SideBar
    menuContent={
      Object {
        "links": Array [
          Object {
            "target": "",
            "text": "",
            "title": "",
            "url": "",
          },
        ],
      }
    }
  />
  <Footer />
</main>
`;
